openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java

```diff
@@ -76,20 +73,53 @@ public LocalJavaRuntime getRuntime(final VersionString versionString, final URL
             final RemoteJavaRuntime remoteJavaRuntime = RemoteRuntimeManager.getInstance().getBestRuntime(versionString, serverEndpoint, vendor, os)
                     .orElseThrow(() -> new RuntimeException("Can not provide or find runtime for version '" +
                             versionString + "', vendor '" + vendor + "' and operating system '" + os + "'"));
+
+            //CHECK IF WE ALREADY HAVE THE SAME RUNTIME (DEACTIVATED & MANAGED) LOCALLY
```

Comment:
If you move this new code block into `installRemoteRuntime()` then you do not need to duplicate it.

Reply:
good point :)